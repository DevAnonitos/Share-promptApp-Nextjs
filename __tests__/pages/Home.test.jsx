import React from "react";
import HomePage from "@/app/page";
import { render, screen, waitFor } from "@testing-library/react";


jest.mock('@/components/Feed', () => ({
  __esModule: true,
  default: () => <div data-testid="feed-mock">Mocked Feed</div>,
}));

describe('HomePage', () => {
  
  it("should render a HomePage Header text", () => {
    render(<HomePage />)
  
    expect(screen.getAllByText("Discover & Share")).toMatchSnapshot();
    expect(screen.getAllByText("Generative AI-Prompts")).toMatchSnapshot();
    
  })

  it('should render the description text', () => {
    const { getByText } = render(<HomePage />);
    expect(
      getByText(/PromptApp is an open-source AI prompting tool for modern world to discover, create and share creative prompts./)
    ).toBeInTheDocument();
    expect(
      getByText(/PromptApp is an open-source AI prompting tool for modern world to discover, create and share creative prompts./)
    ).toMatchSnapshot();
  });

  it('should match the snapshot', () => {
    const { asFragment } = render(<HomePage />);
    expect(asFragment()).toMatchSnapshot();
});

  it("should render the Feed Component", async () => {
    render(<HomePage />)

    expect(screen.getByTestId('feed-mock')).toMatchSnapshot();
  })
})
