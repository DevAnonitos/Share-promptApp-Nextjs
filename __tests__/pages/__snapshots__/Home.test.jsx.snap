// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should match the snapshot 1`] = `
<DocumentFragment>
  <section
    class="w-full flex-center flex-col"
  >
    <h1
      class="head_text text-center "
    >
      Discover & Share
      <br
        class="max-md:hidden"
      />
      <span
        class="orange_gradient text-center"
      >
        Generative AI-Prompts
      </span>
    </h1>
    <p
      class="desc text-center"
    >
      PromptApp is an open-source AI prompting tool for modern world to discover, create and share creative prompts.
    </p>
    <div
      data-testid="feed-mock"
    >
      Mocked Feed
    </div>
  </section>
</DocumentFragment>
`;

exports[`HomePage should render a HomePage Header text 1`] = `
[
  <h1
    class="head_text text-center "
  >
    Discover & Share
    <br
      class="max-md:hidden"
    />
    <span
      class="orange_gradient text-center"
    >
      Generative AI-Prompts
    </span>
  </h1>,
]
`;

exports[`HomePage should render a HomePage Header text 2`] = `
[
  <span
    class="orange_gradient text-center"
  >
    Generative AI-Prompts
  </span>,
]
`;

exports[`HomePage should render the Feed Component 1`] = `
<div
  data-testid="feed-mock"
>
  Mocked Feed
</div>
`;

exports[`HomePage should render the description text 1`] = `
<p
  class="desc text-center"
>
  PromptApp is an open-source AI prompting tool for modern world to discover, create and share creative prompts.
</p>
`;
